geo/src/algorithm/intersects/polygon.rs

```diff
@@ -48,13 +82,40 @@ symmetric_intersects_impl!(Rect<T>, Polygon<T>);
 impl<T> Intersects<Polygon<T>> for Polygon<T>
 where
     T: GeoNum,
+    geo_types::Line<T>: RTreeObject,
 {
     fn intersects(&self, polygon: &Polygon<T>) -> bool {
-        // self intersects (or contains) any line in polygon
-        self.intersects(polygon.exterior()) ||
+        // switch to querying trees above some threshold x: poly_a * poly_b segment count is higher than x
+        if (self.exterior().0.len() + self.interiors().iter().map(|ls| ls.0.len()).sum::<usize>())
+            * (polygon.exterior().0.len()
+                + polygon
+                    .interiors()
+                    .iter()
+                    .map(|ls| ls.0.len())
+                    .sum::<usize>())
+            > MAX_NAIVE_SEGMENTS
+        {
+            let lines_a: Vec<_> = self
+                .exterior()
+                .lines()
+                .chain(self.interiors().iter().flat_map(|ls| ls.lines()))
```

Comment:
Should the last coordinate of the exterior ring connect to the first coordinate of the first interior ring?

Reply:
Oh wait, that's not what's happening here, nevermind!

Reply:
Meep meephttps://github.com/georust/geo/issues/708